docker/clickhouse-builder/assets/arm64.compile.Dockerfile

```diff
@@ -0,0 +1,48 @@
+FROM arm64v8/ubuntu:focal
+
+ARG gosu_ver=1.10
+ARG DEBIAN_FRONTEND=noninteractive
+ARG GIT_TAG="v21.9.2.17-stable"
+
+RUN groupadd -r clickhouse --gid=101 \
+    && useradd -r -g clickhouse --uid=101 --home-dir=/var/lib/clickhouse --shell=/bin/bash clickhouse \
+    && apt-get update \
+    && apt-get install --yes --no-install-recommends \
+        apt-transport-https \
+        ca-certificates \
+        dirmngr \
+        gnupg \
+        locales \
+        wget \
+        tzdata \
+    && wget --progress=bar:force:noscroll "https://github.com/tianon/gosu/releases/download/$gosu_ver/gosu-$(dpkg --print-architecture)" -O /bin/gosu \
+    && chmod +x /bin/gosu
+
+RUN locale-gen en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+ENV TZ UTC
+
+RUN apt-get -y install git cmake python ninja-build
+RUN git clone --branch $GIT_TAG --recursive https://github.com/ClickHouse/ClickHouse.git
+
+RUN apt-get -y install clang-12 build-essential
+ENV CC clang-12
+ENV CXX clang++-12
+
+RUN cd ClickHouse && mkdir build && cd build && cmake ..
+RUN cd ClickHouse/build && ninja -j 2
+RUN mv ClickHouse/build/programs/clickhouse* /usr/bin/
```

Comment:
Can we do a multistage build here and just copy over the binary? Might be simpler that the single binary url approach. And maybe will be simple to push up to dockerhub

Reply:
We could. Two options:
1. Running `docker cp` and copying the binary out of one build container and into another production container
2. Running all the steps in this dockerfile inside one `RUN` command, which also deletes all build and installation artefacts --> the only changes in the volume will be the new build binary.

Just to be able to cache steps when building the Dockerfile (the git clone already takes 10min), I built it now as is.

Reply:
Right, yes 2 is a bit awkward with caching and all.

To clarify, I was thinking something like 

```
FROM arm64v8/ubuntu:focal AS builder
...
RUN cd ClickHouse && mkdir build && cd build && cmake ..
RUN cd ClickHouse/build && ninja -j 2

FROM arm64v8/ubuntu:focal

COPY --from=builder /.../ClickHouse/builder/programes/clickhouse* /usr/bin/

...
```

So we can keep all the caching for the build, but then have a small image to push up to dockerhub

But yeah getting this in as is would be great.

Reply:
Oh, can we do that in one dockerfile? If so, 😍